### PR TITLE
feature: Add Trigger URL CLI for managing and creating trigger URL state

### PR DESCRIPTION
**Pull request by briancain**
This pull request builds off of https://github.com/hashicorp/waypoint/pull/2794 and implements the CLI for managing trigger URL state in Waypoint server. Notice that this pull request is targeted to the pull requests branch rather than main. If the other PR gets merged before this one, I'll retarget off `main` instead.

Some of the UX isn't complete, like returning and show the actual HTTP url to trigger requests with when you create a trigger URL. This work will happen next with the http service implementation. Everything else is here in the pull request though for creating/updating, listing, inspection, and deletion.

### CODE REVIEW

File: internal/cli/main.go [hunk 1]
```diff
@@ -405,6 +405,33 @@ func Commands(
 			}, nil
 		},
 
+		"trigger": func() (cli.Command, error) {
+			return &helpCommand{
+				SynopsisText: helpText["trigger"][0],
+				HelpText:     helpText["trigger"][1],
+			}, nil
+		},
+		"trigger apply": func() (cli.Command, error) {
```
Comment (mitchellh) [inline internal/cli/main.go:414]: "apply" feels a bit weird for triggers, I wonder if we should split into "create" and "update". Triggers don't really feel like some desired state operation... even if internally we support an upsert.

**Review by mitchellh — CHANGES_REQUESTED** [review]
Looks very good! Just a couple questions.

File: internal/cli/main.go (hunk 1 above)
Comment (briancain) [reply internal/cli/main.go:414]: I started down the path with `create` and `update` (see git commit history for that) but wasn't really sure how to handle it cleanly without a bunch of duplicated code or a shared lib for handling operations and flags. If you have ideas I'm all ears!

File: internal/cli/trigger_apply.go [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+	resp, err := c.project.Client().UpsertTrigger(ctx, &pb.UpsertTriggerRequest{
+		Trigger: createTrigger,
+	})
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// TODO(briancain): update output to show trigger URL with wp server attached once http service is implemented
+	c.ui.Output("Trigger %q (%s) has been %s", resp.Trigger.Name, resp.Trigger.Id,
```
[the remark this answers is not in the document]
Comment (briancain) [reply internal/cli/trigger_apply.go:253]: Maybe a `-json` flag that's just a simple map of key values here? Like `{id: "123", url: "http://example.com/123"}`

File: internal/cli/main.go (hunk 1 above)
Comment (mitchellh) [reply internal/cli/main.go:414]: The way we've done it before is make it the same command struct, and just put a `Update bool` on it and then switch on that logic in it. So most of the logic is shared, but ie. if `Update == true` then we require an ID. And we can also change some of the help output that way.

File: internal/cli/trigger_apply.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mitchellh) [reply internal/cli/trigger_apply.go:253]: Possibly, still requires `jq`... speaking as a CI scripter, I'd prefer just a plaintext unix-style output of just what I want for the most common cases.

File: internal/cli/main.go (hunk 1 above)
Comment (briancain) [reply internal/cli/main.go]: Got it! Good idea. Got that updated now.

File: internal/cli/trigger_apply.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (briancain) [reply internal/cli/trigger_apply.go:267]: Makes sense, I updated it to output the ID (and in the future, the URL) at the end of the creation script. Similar to how `waypoint up` shows deployment and release urls!

**Comment by briancain** [comment]
Frontend tests seem to be stuck on yellow, if you click into circle they're green. Just an FYI.

File: internal/cli/trigger_list.go [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+			}
+
+			fmt.Println(str)
+		}
+		return 0
+	}
+
+	c.ui.Output("Trigger URL Configs", terminal.WithHeaderStyle())
+
+	tbl := terminal.NewTable("ID", "Name", "Workspace", "Project", "Application", "Operation", "Description", "Labels", "Last Time Active")
```
Comment (izaaklauer) [inline internal/cli/trigger_list.go:86]: Looks good, but also so wide! I wonder if we should steal an idea from kubectl and have a short version of commands like this, with maybe only the 5 most important columns, and add a `-o wide` option to include all the columns. This is exacerbated by our existing line-wrapping problems: https://github.com/hashicorp/waypoint/issues/1608

Other commands are guilty too though, so it shouldn't block this PR. Just an idea.

Comment (briancain) [reply internal/cli/trigger_list.go:86]: I could see that. I also wouldn't mind punting on it and we could fix it across all commands and use the same flag name 🤔 There's also always the `-json` flag which is nicely formatted and not as wide

**Review by mitchellh — APPROVED** [review]

**Comment by briancain** [comment]
Rebased this branch to fix the merge conflict off main from the server proto, and rebased the upstream parent branch into this one to be up to date.

File: internal/cli/main.go [hunk 4]
```diff
@@ -956,6 +990,17 @@ accesses a Waypoint server requires a token.
...
 	},
 
+	"trigger": {
+		"Trigger URL management",
+		`
+Trigger URL management.
+
+Trigger URLs are used execute to lifecycle operations for Waypoint. They are configured
+ahead of the execution request, and Waypoint server will generate a configuration
+URL for clients to send a request to to execute the operation.
```
Comment (krantzinator) [inline internal/cli/main.go:1000]:
```suggestion
Trigger URLs are used to execute lifecycle operations for Waypoint. They are configured
ahead of the execution request, and Waypoint server will generate a configuration
URL for clients to send a request to execute the operation.
```

**Comment by krantzinator** [comment]
One question as I'm reviewing: do we specify `on Waypoint server` with other CLI commands' help texts? Is it necessary to specify that? It seems superfluous to me.

```
    create     Create and register a trigger URL and register it to Waypoint server
    delete     Delete a trigger URL from Waypoint server
    inspect    Inspect a trigger URL from Waypoint server
    list       List trigger URL configurations on Waypoint server
    update     Update a trigger URL and register it to Waypoint server
```

File: internal/cli/trigger_apply.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (krantzinator) [reply internal/cli/trigger_apply.go:267]: Heh I came to ask for more info. I guess it'll all be in the `list` output so that's fine.

**Review by krantzinator — APPROVED** [review]
Looks good! Excited to do some more in-depth testing once the grpc/http stuff hits. 🎉

**Pull request merged**